docs/source/cuda-reference/host.rst

```diff
@@ -149,8 +154,14 @@ To get the default stream:
 
 .. autofunction:: numba.cuda.default_stream
 
-Streams are instances of :class:`numba.cuda.cudadrv.driver.Stream`:
+To get the default stream with an explicit choice of whether it is the legacy
+or per-thread default stream:
 
-.. autoclass:: numba.cuda.cudadrv.driver.Stream
-   :members: synchronize, auto_synchronize
+.. autofunction:: numba.cuda.legacy_default_stream
+
+.. autofunction:: numba.cuda.per_thread_default_stream
+
+To construct a Numba ``Stream`` object using a stream allocated elsewhere, the
+``external_stream`` function is provided:
 
+.. autofunction:: numba.cuda.external_stream
```

Comment:
I think it'd be nice to explicitly state that it's users' responsibility to manage the lifetime of any external stream. 

ref: cupy/cupy#3141

Reply:
Good point - I had overlooked this.

Reply:
@leofang I've now updated the documentation.

Reply:
Thanks, Graham!